drivers/counter/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ zephyr_library_sources_ifdef(CONFIG_COUNTER_INFINEON_TCPWM counter_infineon_tcpw
 zephyr_library_sources_ifdef(CONFIG_COUNTER_ITE_IT51XXX counter_ite_it51xxx.c)
 zephyr_library_sources_ifdef(CONFIG_COUNTER_ITE_IT8XXX2 counter_ite_it8xxx2.c)
 zephyr_library_sources_ifdef(CONFIG_COUNTER_MAXIM_DS3231 maxim_ds3231.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MCHP_G1_TCC counter_mchp_tcc_g1.c)
```

Comment:
Now, you are adding a driver. Then the commit brief should tell us that.

this
`drivers: microchip: sam_e54_xpro: Add counter g1 driver with tcc`
should be something like
`drivers: counter: microchip: Add TCC G1 driver`

a driver of type counter from vendor MCHP ...